lib/megaLogger.js

```diff
@@ -252,55 +232,48 @@
      * Send the passed `arguments` to the specific `level` logger
      *
      * @param level {number}
-     * @param arguments {*}
+     * @param args {*}
      * @private
      */
-    MegaLogger.prototype._log = function(level, arguments) {
-        var self = this;
-
-        var levelName = MegaLogger._intToLevel(level);
-        var clr = self.options.levelColors[levelName];
-        var logStyle = "color: " + "white" + "; background-color: " +  clr + "; padding-left: 1px; padding-right: 1px;";
+    MegaLogger.prototype._log = function(level, args) {
+        var options = this.options;
 
-
-        var args = [
-            (self.options.colorsEnabled ? "%c" : "") + self.options.dateFormatter(new Date()) + " - " + self._getLoggerPath() + " - " + levelName,
-            (self.options.colorsEnabled ? logStyle : "")
-        ];
-
-        args = args.concat(arguments);
-        if(self.options.dereferenceObjects) {
-            args = JSON.parse(JSON.stringify(args));
+        // check min log level
+        if (level < options.minLogLevel()) {
+            return;
         }
 
-        if(self.options.isEnabled === true || (isfunction(self.options.isEnabled) && self.options.isEnabled())) {
+        if (options.isEnabled === true || (isfunction(options.isEnabled) && options.isEnabled())) {
+            var logStyle = "";
+            var levelName = _intToLevel[level] || "unknown";
+            var logLine = options.dateFormatter(new Date()) + " - " + this._getLoggerPath() + " - " + levelName;
 
+            if (options.colorsEnabled) {
+                var clr = options.levelColors[levelName];
+                logStyle = "color: white; background-color: " +  clr + "; padding-left: 1px; padding-right: 1px;";
+                logLine = "%c" + logLine;
+            }
 
-            var txtMsg = args.join(" ");
-            var muted = false;
-            self.options.muteList().forEach(function(v) {
-                var r = new RegExp(v);
-                if(r.test(txtMsg)) {
-                    muted = true;
-                    return false; // break;
-                }
-            });
-
-            if(muted) {
-                return;
+            args = [logLine, logStyle].concat(args);
+            if (options.dereferenceObjects) {
+                args = JSON.parse(JSON.stringify(args));
             }
 
-            if(level < self.options.minLogLevel()) { // check min log level
-                return;
+            var muteList = options.muteList();
+            if (muteList.length) {
+                var txtMsg = args.join(" ");
+                if (muteList.some(function(v) { return RegExp(v).test(txtMsg); })) {
+                    return;
+                }
```

Comment:
This is just a tiny improvement for the `muteList`, relying on Array's `some` over `forEach`...